backend/src/main/java/cloudflight/integra/backend/trip/TripServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package cloudflight.integra.backend.trip;
+
+import cloudflight.integra.backend.coffeeShop.CoffeeShopRepository;
+import cloudflight.integra.backend.poi.DBPOIRepository;
+import cloudflight.integra.backend.restaurant.DBRestaurantRepository;
+import cloudflight.integra.backend.spa.DBSpaRepository;
+import cloudflight.integra.backend.user.DBUserRepository;
+import cloudflight.integra.backend.user.User;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TripServiceImpl implements TripService {
+    private final DBTripRepository tripRepository;
+    private final DBUserRepository userRepository;
+    private final DBPOIRepository poiRepository;
+    private final DBRestaurantRepository restaurantRepository;
+    private final CoffeeShopRepository coffeeShopRepository;
+    private final DBSpaRepository spaRepository;
+
+    public TripServiceImpl(
+            DBTripRepository tripRepository,
+            DBUserRepository userRepository,
+            DBPOIRepository poiRepository,
+            DBRestaurantRepository restaurantRepository,
+            CoffeeShopRepository coffeeShopRepository,
+            DBSpaRepository spaRepository) {
+        this.tripRepository = tripRepository;
+        this.userRepository = userRepository;
+        this.poiRepository = poiRepository;
+        this.restaurantRepository = restaurantRepository;
+        this.coffeeShopRepository = coffeeShopRepository;
+        this.spaRepository = spaRepository;
+    }
+
+    @Override
+    public Trip addTrip(Trip trip) {
+        return tripRepository.save(trip);
+    }
+
+    @Override
+    public List<Trip> addTrips(List<Trip> trips) {
+        return tripRepository.saveAll(trips);
+    }
+
+    @Override
+    public Trip updateTrip(Long id, Trip trip) {
+        Trip existingTrip =
+                tripRepository.findById(id).orElseThrow(() -> new RuntimeException("Trip not found with id: " + id));
+        existingTrip.setName(trip.getName());
+        existingTrip.setCity(trip.getCity());
+        existingTrip.setDays(trip.getDays());
+        existingTrip.setPrice(trip.getPrice());
+        existingTrip.setUser(trip.getUser());
+        return tripRepository.save(existingTrip);
+    }
+
+    @Override
+    public void deleteTrip(Long id) {
+        if (!tripRepository.existsById(id)) {
+            throw new RuntimeException("Trip not found with id: " + id);
+        }
+        tripRepository.deleteById(id);
```

Comment:
This is not standard across entities. Deletion of an inexistent entity should be allowed as it doesn't throw an error.
```suggestion
        tripRepository.deleteById(id);
```